src/lib/middleware.jsx

```diff
@@ -16,7 +16,7 @@ import { history } from '../lib/helpers';
 export const isLoggedIn = currentUser =>
   new Promise(resolve => {
     if (currentUser && currentUser.address) resolve();
-    else history.goBack();
+    else history.push('/');
```

Comment:
This is super ugly to have it in the isLoggedIn user. Could this just reject the promise and then the functionality be handled outside? I may even make sense to inform the user that they were accessing protected URL and should login first. What do you think?

Reply:
Yes, I agree that makes a lot of sense, but in the meantime this is a better solution